aws-java-sdk-auth-userpools/src/main/java/com/amazonaws/mobile/auth/userpools/GluonView.java

```diff
@@ -0,0 +1,142 @@
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.amazonaws.mobile.auth.userpools;
+
+import com.gluonhq.charm.glisten.application.MobileApplication;
+import com.gluonhq.charm.glisten.application.ViewStackPolicy;
+import com.gluonhq.charm.glisten.control.AppBar;
+import com.gluonhq.charm.glisten.mvc.View;
+import com.gluonhq.charm.glisten.visual.MaterialDesignIcon;
+import java.util.Collection;
+import java.util.MissingResourceException;
+import java.util.ResourceBundle;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javafx.application.Platform;
+import javafx.scene.Node;
+import javafx.scene.control.ProgressBar;
+import javafx.scene.layout.VBox;
+
+public class GluonView extends View {
+
+    static final Logger LOG = Logger.getLogger(GluonView.class.getName());
+
+    private static final ResourceBundle BUNDLE;
+    static {
+        BUNDLE = ResourceBundle.getBundle("com/amazonaws/mobile/auth/userpools/userpools");
+    }
+    
+    private final VBox center;
+    
+    public GluonView() {
+        getStyleClass().add("aws");
+        
+        center = new VBox();
+        center.getStyleClass().add("center");
+        setCenter(center);
+        
+        setOnShowing(e -> showProgressBar(false));
+    }
+
+    final void addNodes(Collection<Node> nodes) {
+        center.getChildren().addAll(nodes);
+    }
+    
+    final void addNodes(Node... nodes) {
+        center.getChildren().addAll(nodes);
+    }
+
+    @Override
+    protected void updateAppBar(AppBar appBar) {
+        appBar.setNavIcon(MaterialDesignIcon.CHEVRON_LEFT.button(e -> {
+            getApplication().switchToPreviousView();
+        }));
+    }
+
+    
+    @Override
+    public String getUserAgentStylesheet() {
+        return GluonView.class.getResource("aws-style.css").toExternalForm();
+    }
+    
+    static String getString(String value) {
+        try {
+            return BUNDLE.getString(value);
+        } catch (MissingResourceException ex) {
+            LOG.log(Level.SEVERE, null, ex);
+        }
+        return "";
+    }
+    
+    static void switchView(final String viewName) {
+        if (Platform.isFxApplicationThread()) {
+            MobileApplication.getInstance().switchView(viewName, ViewStackPolicy.SKIP);
+        } else {
+            Platform.runLater(() -> {
+                MobileApplication.getInstance().switchView(viewName, ViewStackPolicy.SKIP);
+            });
+        }
+    }
+    
+    static void showProgressBar(boolean value) {
+        final AppBar appBar = MobileApplication.getInstance().getAppBar();
+        if (Platform.isFxApplicationThread()) {
+            appBar.setProgress(value ? ProgressBar.INDETERMINATE_PROGRESS : 0);
+            appBar.setProgressBarVisible(value);
+        } else {
+            Platform.runLater(() -> {
+                appBar.setProgress(value ? ProgressBar.INDETERMINATE_PROGRESS : 0);
+            appBar.setProgressBarVisible(value);/**
```

Comment:
The header here is in a strange location.

Reply:
Good catch! Removed.